### PR TITLE
keystore: export `Keystore`

### PR DESCRIPTION
**Pull request by xenowits**
Exports `Keystore` and `Encrypt()` from keystore package.

category: refactor
ticket: #1503

### CODE REVIEW

File: eth2util/keystore/keystore.go [hunk 1]
```diff
@@ -169,31 +142,31 @@ type keystore struct {
 	Version     uint                   `json:"version"`
 }
 
-// encrypt returns the secret as an encrypted keystore using pbkdf2 cipher.
-func encrypt(secret *bls_sig.SecretKey, password string, random io.Reader,
+// Encrypt returns the secret as an encrypted Keystore using pbkdf2 cipher.
+func Encrypt(secret *bls_sig.SecretKey, password string, random io.Reader,
```
Comment (xenowits) [inline eth2util/keystore/keystore.go:146]: https://github.com/ObolNetwork/charon/pull/1622#discussion_r1071268819

File: cmd/createcluster.go [hunk 2]
```diff
@@ -749,3 +750,41 @@ func verifyConnection(ctx context.Context, addr string) error {
...
 }
+
+// keymanagerReq represents the keymanager API request body for POST request. Refer: https://ethereum.github.io/keymanager-APIs/#/Local%20Key%20Manager/importKeystores
+type keymanagerReq struct {
+	Keystores []keystore.Keystore `json:"keystores"`
+	Passwords []string            `json:"passwords"`
+}
+
+// keymanagerReqBody constructs a keymanagerReq using the provided secrets and returns it.
+func keymanagerReqBody(secrets []*bls_sig.SecretKey) (keymanagerReq, error) {
```
Comment (corverroos) [inline cmd/createcluster.go:761]: suggest moving to its own file since it will be used in dkg as well

**Review by corverroos — APPROVED** [review]

File: cmd/keymanager.go [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+)
+
+// KeymanagerReq represents the keymanager API request body for POST request. Refer: https://ethereum.github.io/keymanager-APIs/#/Local%20Key%20Manager/importKeystores
+type KeymanagerReq struct {
+	Keystores []keystore.Keystore `json:"keystores"`
+	Passwords []string            `json:"passwords"`
+}
+
+// KeymanagerReqBody constructs a KeymanagerReq using the provided secrets and returns it.
+func KeymanagerReqBody(secrets []*bls_sig.SecretKey) (KeymanagerReq, error) {
```
Comment (xenowits) [inline cmd/keymanager.go:35]: This is public so that dkg package can use it when supporting keymanager

Comment (corverroos) [reply cmd/keymanager.go:35]: ah right, dkg isn't in cmd package, then it needs its own package, suggest "/eth2util/keymanager"

**Review by corverroos — CHANGES_REQUESTED** [review]
Forgot DKG is in different package, so this needs its own package, suggest `eth2util/keymanager`

**Comment by xenowits** [comment]
Closing this in favour of https://github.com/ObolNetwork/charon/pull/1657

**Pull request closed without merging**